addons/web_tour/static/src/js/tour_manager.js

```diff
@@ -282,6 +282,16 @@ return core.Class.extend(mixins.EventDispatcherMixin, ServicesMixin, {
             }
         }
     },
+    /**
+     * @param {String} tourUrl
+     * @returns {boolean} true if the browser should be redirected to the
+     *          tour's url.
+     */
+    _shouldRedirect(tourUrl) {
+        const windowUrl = new URL(window.location.href);
+        windowUrl.searchParams.delete('watch');
```

Comment:
What's that? Need comments :)

Reply:
Before this commit, the tour manager only checks if the tour is defined with a url. If so, it redirects the browser to the tour's url. This commit adds an extra check on the current location of the browser, to avoid redirecting it if it is already at the right location. I guess this extra check could be implemented before 16.0.

For the watch parameter being removed from the check, I found out as I was launching the python tours in watch mode (I usually set the default value of the watch param to true, but there is probably a better, more conventional way to do it).
```py
def browser_js(self, url_path, code, ready='', login=None, timeout=60, cookies=None, error_checker=None, watch=True, **kw):
```
The python test suite adds a parameter when starting the browser for the test
![image](https://user-images.githubusercontent.com/58561718/202663499-284534de-49e9-457b-b30a-ae05a15ea553.png)
In order to avoid reloading the browser in that configuration, the watch parameter was filtered out

Reply:
> For the watch parameter being removed from the check, I found out as I was launching the python tours in watch mode (I usually set the default value of the watch param to true, but there is probably a better, more conventional way to do it).

I saw the code before but I am still wondering how you found out about it ^^ But whatever, I guess unrelated to this PR, could be considered in older versions, that's true, that seems like a nice perf improvement for the frontend tours :open_mouth: :+1:  For the backend tours, there is not that problem as changing URL to another backend URL does not reload the page (there is the `do_before_unload` stuff.